README.md

```diff
@@ -1,2 +1,107 @@
 # fastify-swagger
-Swagger documentation generator for Fastify
+[![js-standard-style](https://img.shields.io/badge/code%20style-standard-brightgreen.svg?style=flat)](http://standardjs.com/)  [![Build Status](https://travis-ci.org/fastify/fastify-swagger.svg?branch=master)](https://travis-ci.org/fastify/fastify-swagger)
+
+Swagger documentation generator for Fastify.  
+It uses the schemas you declare in your routes to generate a swagger compliant doc.
+
+<a name="install"></a>
+## Install
+```
+npm i fastify-swagger --save
+```
+
+<a name="usage"></a>
+## Usage
+Add it to your project with `register` and pass it some basic options, then call the `swagger` api and you are done!
+
+*Note that currently only the generation of yaml/json files is supported, the swagger UI will not be generated.*
+```js
+const fastify = require('fastify')()
+
+fastify.register(require('fastify-swagger'), {
+  swagger: {
+    info: {
+      title: 'Test swagger',
+      description: 'testing the fastify swagger api',
+      version: '0.1.0'
+    },
+    host: 'localhost',
+    schemes: ['http'],
+    consumes: ['application/json'],
+    produces: ['application/json']
+  }
+})
+
+fastify.post('/some-route', {
+  schema: {
+    description: 'post some data',
+    tags: ['user', 'code'],
+    summary: 'qwerty',
+    payload: {
+      type: 'object',
+      properties: {
+        hello: { type: 'string' },
+        obj: {
+          type: 'object',
+          properties: {
+            some: { type: 'string' }
+          }
+        }
+      }
+    },
+    out: {
+      description: 'Succesful response',
+      code: 201,
+      type: 'object',
+      properties: {
+        hello: { type: 'string' }
+      }
+    }
+  }
+}, (req, reply) => {})
+
+fastify.ready(err => {
+  if (err) throw err
+  fastify.swagger()
```

Comment:
Why this?
Can this plugin attach to `ready` avvio event to generate the swagger?

I don't remember if `fastify` allow it, but what happen if a route is attached after the start up? Can be rebuild with a `{ force: true }`?

Reply:
`fastify.swagger()` is a synchronous api, we can't use `.ready` inside as we did for `inject`, otherwise send the result inside a route will be painful.

> I don't remember if fastify allow it, but what happen if a route is attached after the start up? Can be rebuild with a { force: true }?

I didn't try, but I don't think it's a good idea. Furthermore, which is the use case?

Reply:
> I didn't try, but I don't think it's a good idea. Furthermore, which is the use case?

no one. Please update the doc